src/Microsoft.ML.Data/Training/LearnerInputBase.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Data.DataView;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.Data.IO;
+using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Internal.Calibration;
+
+namespace Microsoft.ML.Trainers
+{
+    /// <summary>
+    /// The base class for all learner inputs.
+    /// </summary>
+    [TlcModule.EntryPointKind(typeof(CommonInputs.ITrainerInput))]
+    public abstract class LearnerInputBase
```

Comment:
>Learner [](start = 26, length = 7)

We discuss this today, and all `LearnerSomething` should become `TrainerSomething` `LearnerInputBase` ->`TrainerInputBase`

Reply:
Which can be scope of other PR

---
In reply to: [260525998](https://github.com/dotnet/machinelearning/pull/2748#discussion_r260525998) [](ancestors = 260525998)

Reply:
this done

---
In reply to: [260527749](https://github.com/dotnet/machinelearning/pull/2748#discussion_r260527749) [](ancestors = 260527749,260525998)